modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3790,6 +3795,62 @@ def __convert_by(by):
 
         return unstacked
 
+    def _pivot_table_tree_reduce(
+        self, grouper, aggfunc, drop_column_level, fill_value, dropna, to_unstack=None
+    ):
+        """
+        Build a pivot table using TreeReduce implementation.
+
+        Parameters
+        ----------
+        grouper : PandasQueryCompiler
+            QueryCompiler holding columns to group on.
+        aggfunc : str
+            Aggregation to perform against the values of the pivot table. Note that ``GroupbyReduceImpl``
+            has to be able to build implementation for this aggregation.
+        drop_column_level : bool
+            Whether to drop the top level of the columns.
+        fill_value : object
+            Fill value for None values in the result.
+        dropna : bool
+            Whether to drop NaN columns.
+        to_unstack : list, optional
+            A list of column names to pass to the `.unstack()` when building the pivot table.
+            If `None` was passed perform regular transpose instead of unstacking.
+
+        Returns
+        -------
+        PandasQueryCompiler
+            A query compiler holding a pivot table.
+        """
+
+        def make_pivot_table(df):
+            if df.index.nlevels > 1 and to_unstack is not None:
+                df = df.unstack(level=to_unstack)
+            if drop_column_level and df.columns.nlevels > 1:
+                df = df.droplevel(0, axis=1)
+            if dropna:
+                df = df.dropna(axis=1, how="all")
+            if fill_value is not None:
+                df = df.fillna(fill_value, downcast="infer")
+            return df
+
+        result = GroupbyReduceImpl.build_qc_method(
+            aggfunc, finalizer_fn=make_pivot_table
+        )(
+            self,
+            by=grouper,
+            axis=0,
+            groupby_kwargs={},
+            agg_args=(),
+            agg_kwargs={},
+            drop=True,
+        )
```

Comment:
Why not?
```suggestion
        result = GroupbyReduceImpl.build_qc_method(aggfunc)(
            self,
            by=grouper,
            axis=0,
            groupby_kwargs={},
            agg_args=(),
            agg_kwargs={},
            drop=True,
        )
        result = make_pivot_table(result)
```

Reply:
we want the `make_pivot_table` function to be executed inside of a kernel that's running the groupby itself, in your suggestion the `result` is a query compiler, meaning that applying the `make_pivot_table()` to it would result into another kernel to be submitted, which is not that performance efficient as a single-kernel case

Reply:
performance reason, I see